Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

Run npm install
npm warn old lockfile
npm warn old lockfile The package-lock.json file was created with an old version of npm,
npm warn old lockfile so supplemental metadata must be fetched from the registry.
npm warn old lockfile
npm warn old lockfile This is a one-time fix-up, please be patient...
npm warn old lockfile
npm warn deprecated lodash.get@4.4.2: This package is deprecated. Use the optional chaining (?.) operator instead.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.1.3: Glob versions prior to v9 are no longer supported
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported

added 273 packages, and audited 274 packages in 10s

41 packages are looking for funding
  run `npm fund` for details

1 high severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.

> filetransform@1.0.0 build
> tsc

src/variableSubstitution.ts(36,63): error TS234[5](https://github.com/PavanMudigondaTR/variable-substitution/actions/runs/14382779653/job/40330534255#step:3:6): Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.
src/variableSubstitution.ts(4[6](https://github.com/PavanMudigondaTR/variable-substitution/actions/runs/14382779653/job/40330534255#step:3:7),130): error TS2322: Type 'string' is not assignable to type 'BufferEncoding'.
src/variableSubstitution.ts(6[7](https://github.com/PavanMudigondaTR/variable-substitution/actions/runs/14382779653/job/40330534255#step:3:8),62): error TS2322: Type 'string' is not assignable to type 'BufferEncoding'.
src/variableSubstitution.ts([8](https://github.com/PavanMudigondaTR/variable-substitution/actions/runs/14382779653/job/40330534255#step:3:9)1,120): error TS2322: Type 'string' is not assignable to type 'BufferEncoding'.

--- a/src/variableSubstitution.ts
+++ b/src/variableSubstitution.ts
@@ -33,7 +33,7 @@ export class VariableSubstitution {
             for(let file of matchedFiles) {
                 let fileBuffer: Buffer = fs.readFileSync(file);
                 let fileEncodeType = fileEncoding.detectFileEncoding(file, fileBuffer);
-                let fileContent: string = fileBuffer.toString(fileEncodeType.encoding);
+                let fileContent: string = fileBuffer.toString(fileEncodeType.encoding as BufferEncoding);
                 if(fileEncodeType.withBOM) {
                     fileContent = fileContent.slice(1);
                 }
@@ -43,7 +43,7 @@ export class VariableSubstitution {
                     let jsonObject = this.fileContentCache.get(file);
                     let isJsonSubstitutionApplied = jsonSubsitution.substituteJsonVariable(jsonObject, EnvTreeUtility.getEnvVarTree());
                     if(isJsonSubstitutionApplied) {
-                        fs.writeFileSync(file, (fileEncodeType.withBOM ? '\uFEFF' : '') + JSON.stringify(jsonObject, null, 4), { encoding: fileEncodeType.encoding });
+                        fs.writeFileSync(file, (fileEncodeType.withBOM ? '\uFEFF' : '') + JSON.stringify(jsonObject, null, 4), { encoding: fileEncodeType.encoding as BufferEncoding });
                         console.log(`Successfully updated file: ${file}`);
                     }
                     else {
@@ -64,7 +64,7 @@ export class VariableSubstitution {
                             core.debug('Substituting original value in place of temp_name: ' + replacableTokenValue);
                             domContent = domContent.split(replacableTokenValue).join(xmlSubstitution.replacableTokenValues[replacableTokenValue]);
                         }
-                        fs.writeFileSync(file, domContent, { encoding: fileEncodeType.encoding });
+                        fs.writeFileSync(file, domContent, { encoding: fileEncodeType.encoding as BufferEncoding });
                         console.log(`Successfully updated file: ${file}`);
                     }
                     else {
@@ -78,7 +78,7 @@ export class VariableSubstitution {
                     let yamlObject = this.fileContentCache.get(file);
                     let isYamlSubstitutionApplied = jsonSubsitution.substituteJsonVariable(yamlObject, EnvTreeUtility.getEnvVarTree());
                     if(isYamlSubstitutionApplied) {
-                        fs.writeFileSync(file, (fileEncodeType.withBOM ? '\uFEFF' : '') + yaml.safeDump(yamlObject), { encoding: fileEncodeType.encoding });
+                        fs.writeFileSync(file, (fileEncodeType.withBOM ? '\uFEFF' : '') + yaml.safeDump(yamlObject), { encoding: fileEncodeType.encoding as BufferEncoding });
                         console.log(`Successfully updated config file: ${file}`);
                     }
                     else {

